### DISCUSSION TITLE
Corel X5 + <image>

### DISCUSSION THREAD

**Discussion by ejtipi** [fabricjs/fabric.js#7113] [Q&A]
## Version
4.3.1

## Test File
https://designbox.pro/_misc/60c130ed03e16-1623273938.svg

This file open's properly in Opera, Chrome, IE, Edge, Firefox, Adobe Illustrator, Inkscape, Adobe Photoshop but not in Corel Draw X5.

I've simply added image to canvas like:

```
fabric.Image.fromURL(data, function (img) {
	var oImg = img.set({left: 30, top: 30, angle: 0});
	canvas.add(oImg).renderAll();
});
```

Element from <image> tag opened in Corel is very small - rest of file is ok.

![corel-image](https://user-images.githubusercontent.com/25153797/121676198-a66cbc00-cab4-11eb-826d-283fe63f98a3.png)

I realize this is a Corel bug, but maybe someone has found a workaround to open files like this properly?

**Comment by asturur** [comment]
hi @ejtipi there is really little going on in this SVG.
You can try 2 things:
1- remove all the inline style props from the img and see if it looks correct
2. remove the transform from the img tag, and multiply the height/width by 0.08 ( the transforms value ) and see if it looks good.

I would not know what else to try.

**Reply by ejtipi** [reply]
Second idea works!

Do you have any idea how it can be done automatically? If this would affect the speed of the browser, maybe it would be better to do it on the server side in PHP?

**Reply by asturur** [reply]
is not an issue with speed but compatibility
That svg transform is doing scale + rotate + translate and i m not sure how you can reverse that in a way the image is correctly positioned.

if instead of the matrix you add

`transform="scale(0.08 0.08)"` how does it look?

**Reply by ejtipi** [reply]
`transform="scale(0.08 0.08)"`
Does not work :/

EDIT: If i move `transform="matrix(0.08 0 0 0.08 214.27 212.5)"` from `<g>` tag to `<image>` tag it opens correctly! ...and if i put transform scale directly to `<image>` tag it works too :)

EDIT2: Attached in topic SVG file is empty at the beginning, than user adds their own elements - `<g>` tag must be added by Fabric.js. Is there a way to avoid this and put transform directly to image tag?

**Reply by asturur** [reply]
so the issue is that clipPath then would stop to work. This is a big issue on corel side? how old is x5?
So to solve it in an elegant way, maybe, you can try to use the reviver option from the SVG option.

That should allow you to pass in a function that gets run on each SVG string produced
Thas should allow to process the group+image block when is ready.
You can have a regexp that takes the transfrom and put it over the image.

**Reply by ejtipi** [reply]
Corel X5 was released in 2010, however, it works best with an engraving machine.

In Fabric.js 3.6.3 i can simple modify _createBaseSVGMarkup function to avoid wraping image tags in g tags and put transform directly to image tag (i've read in documentation that g tags are for grouping only) like `if(this.type !== 'image') ...` - is there a way to do this in 4.3.1?

**Reply by asturur** [reply]
what is wrong with the reviver function? seems like an easy way to go, 1 regex to replace, 1 regex to add and you don't have to bother with which version of fabricJS are you using.
No?

**Reply by ejtipi** [reply]
Looks promising ;) What do you think?

```
    var tempSVG = canvas.toSVG(null, function(tempSVG) {
	   if(tempSVG.includes('<image')) {
		   var imgTransform = tempSVG.match(/<g.*?transform=['"](.*?)['"]/);
		   if(imgTransform[1]) {

			   tempSVG = tempSVG.replace('<image', '<image transform="'+imgTransform[1]+'"');
			   tempSVG = tempSVG.replace(/<g.*?transform=['"](.*?)['"]/gm, '<g');
   
		   }
	   }
    });
```

**Reply by asturur** [reply]
i think it can work. There could be some weird edge cases if something is between g and transform,  im not sure if you can use the capture groups to be safer.

wouldn't be safer to:

get the transform in imgTransform.

replace in the string imgTrasnform with emptyString ( leaving transform="" on the group )

then do 

```
tempSVG = tempSVG.replace('<image', '<image transform="'+imgTransform[1]+'"');
```

just because i m not sure if this is safe

```
tempSVG = tempSVG.replace(/<g.*?transform=['"](.*?)['"]/gm, '<g');
```
